airsonic-main/src/main/java/org/airsonic/player/util/LegacyHsqlUtil.java

```diff
@@ -71,13 +72,19 @@ public static boolean isHsqldbDatabaseUpgradeNeeded() {
         // Check the database driver version
         String driverVersion = null;
         try {
-            Driver driver = DriverManager.getDriver(SettingsService.getDefaultJDBCUrl());
+            Driver driver = (Driver) Class.forName("org.hsqldb.jdbc.JDBCDriver", true, LegacyHsqlUtil.class.getClassLoader()).getDeclaredConstructor().newInstance();
```

Comment:
Because we have to be aware of class loading. We would need to load the caller class using the same class loader as the driver (or a child of that class loader), or SystemClassLoader.